proposals/lip-0043.md

```diff
@@ -280,161 +273,136 @@ This command does not have an extra fee, i.e., `extra fee = 0`.
-    "ownName",
-    "mainchainValidators",
-    "signature",
-    "aggregationBits"
-  ]
 }
 ```
 
 #### Verification
 
 Let `trs` be a transaction with module ID `MODULE_ID_INTEROPERABILITY` and command ID `COMMAND_ID_MAINCHAIN_REG` to be verified. The set of validity rules to validate `trs.params` are:
 
-* The `trs.params.mainchainValidators.keys` array must contain 101 elements (exact number of active delegates on Lisk mainchain.
-* Every element in the `trs.params.mainchainValidators.keys` array has to be unique.
-* Elements in the `trs.params.mainchainValidators.keys` array have a length of 48 bytes.
-* The `trs.params.mainchainValidators.keys` array must be ordered lexicographically.
-* The `trs.params.mainchainValidators.weights` array must contain the same number of elements as `trs.params.mainchainValidators.keys`.
-* The elements in the `trs.params.mainchainValidators.weights` array are integers with value equal to 1.
-* The value of the `trs.params.mainchainValidators.certificateThreshold` property has to be 68.
+* The `trs.params.ownChainID` property has to be an integer less than or equal to `MAX_UINT32`. 
+* The `trs.params.ownName` property has to contain only characters from the set `[a-z0-9!@$&_.]` and has to be at most `MAX_LENGTH_NAME` characters long.
+* Rules for `trs.params.mainchainValidators` array:
+    *   The array must contain 101 elements (exact number of active delegates on Lisk mainchain).
+    *   The array must be ordered lexicographically by `blsKey` property.
+    *   The `blsKey` property of each element has a length of 48 bytes.
+    *   The `blsKey` property of each element is unique within the array.
+    *   The `bftWeight` property of each element is equal to 1.
 * The properties `trs.params.aggregationBits` and `trs.params.signature` are validated as follows:
   * The function `verifyWeightedAggSig(keyList, tag, netID, aggregationBits, signature, weights, certificateThreshold, message)` specified in the [LIP 0038][BLSkeys], must return `VALID` where:
-    * The `keyList` property is an array containing `getValidatorAccount(address).blsKey` for every address in `currentValidators = getGeneratorList()`, where `getValidatorAccount` and `getGeneratorList` are functions exposed by the [validators module][validatorsModule].
-    * The `tag` is equal to `TAG_CHAINREG_MESSAGE`.
+    * Let `validators` be the array of objects as 4th output of the `getBFTParameters` function exposed by the [BFT module][BFTModule], i, e., `( , , , validators, ) = getBFTParameters(currentHeight)`, where `currentHeight` is the current height of the chain as defined for the BFT module. Then:
+       *  The `keyList` argument is an array containing `getValidatorAccount(address).blsKey`, for every `address` element in `validators`. The `getValidatorAccount` function is exposed by the [Validators module][validatorsModule].
+       *  The `weights` argument is an array containing every `bftWeight` element in `validators`.
```

Comment:
Maybe more compactly and clearly we could write:
```
keyList = [ validatorsModule.getValidatorAccount(validator.address).blsKey for validator in validators]
weights = [ validator.bftWeight for validator in validators]
```
It is also important that the order in both arrays is the same and corresponds to the `aggregationBits` argument (if you do not re-sort, you would use the same order as in the BFT module. This means the order is lexicographically by address).

Reply:
thanks, changed as suggested and imposing ordering respect to BLS keys in both arrays.

Reply:
Okay, makes sense.